vyper/codegen/stmt.py

```diff
@@ -277,14 +278,21 @@ def _parse_For_range(self):
             rounds_bound = rounds
 
         # Type 3 for, e.g. for i in range(x, x + 10): ...
-        else:
+        elif "bound" not in kwargs:
             arg1 = self.stmt.iter.args[1]
             rounds = self._get_range_const_value(arg1.right)
             start = Expr.parse_value_expr(arg0, self.context)
             _, hi = start.typ.int_bounds
             start = clamp("le", start, hi + 1 - rounds)
             rounds_bound = rounds
 
+        # Type 4 for, e.g. for i in range(1, x, bound=4): ...
+        else:
+            start = Expr.parse_value_expr(arg0, self.context)
+            end = Expr.parse_value_expr(self.stmt.iter.args[1], self.context)
+            rounds = IRnode.from_list(["sub", end, start], typ=iter_typ)
```

Comment:
i think an underflow check is needed here

Reply:
I am trying to reproduce an underflow in tests. However [this test](https://github.com/DanielSchiavini/vyper/blob/a6fb6a059ebda8403044de05b36ca925cb664858/tests/functional/codegen/features/iteration/test_for_range.py#L68) seems to cover the possibility of underflow. Am I missing something?

Reply:
yea i think that happens to be caught because of the range check in `repeat`. `-1` in two's complement is much greater than 6. but you can construct a start+end/bound which will pass the test, maybe something like `end - start == -5` and `bound = 4`

Reply:
I could not reproduce the underflow, but I've added a runtime check